### PR TITLE
WIP: Add config option to enable/disable default fuzzy search

### PR DESCRIPTION
**Pull request by 6543**
explicit enable or disable it should be an UI option

---
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

File: docs/content/administration/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -467,6 +467,7 @@ relation to port exhaustion.
 
 ## Indexer (`indexer`)
 
+- `INDEXER_DEFAULT_FUZZY`: **true**: If the indexer type supports fuzzy search use it by default. Is used by issue- and repo-indexer alike.
```
Comment (silverwind) [inline docs/content/administration/config-cheat-sheet.en-us.md:470]: Maybe help the user and list a few known ones that support it? Or maybe ones that don't?

**Comment by silverwind** [comment]
Why offload the option to the user? Don't we know which ones support it?

File: docs/content/administration/config-cheat-sheet.en-us.md (hunk 1 above)
Comment (silverwind) [reply docs/content/administration/config-cheat-sheet.en-us.md:470]: Ah, I actually misread.

**Comment by silverwind** [comment]
1. Would reword: "Whether to use futty search by default in the issue, if the indexer supports it".
2. Can we cut out the `by default`? What does it refer to?
3. I think the UI dropdown's default option will need to be adjusted.

**Comment by 6543** [comment]
to 2. I want to have an option in the UI later on that enable/disable fuzzyness and to not have the key in settings changed, i named it, so it can server as default value ... later on

anyway, currently if you switch from bleach to meilisearch, you will get a lot of unrelated issues in issue search ...

**Comment by silverwind** [comment]
> to 2. I want to have an option in the UI later on that enable/disable fuzzyness and to not have the key in settings changed, i named it, so it can server as default value ... later on

The code search has this already, but I think for issue search we should support is via keyword `mode:fuzzy` or something.

image

> anyway, currently if you switch from bleach to meilisearch, you will get a lot of unrelated issues in issue search ...

Yeah, sounds like a big. We should strive to configure indexers so they yield similar results.

**Comment by 6543** [comment]
well be addressed by #29701
